### ISSUE TITLE
Yolov12 for instance segmentation task

### ISSUE THREAD

**Issue by mrohit01** [ultralytics/ultralytics#19398]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

How can I train Yolov12 for instance segmentation model?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @mrohit01, thank you for your interest in Ultralytics 🚀! We recommend visiting the [Docs](https://docs.ultralytics.com/) for new users where you can find many helpful resources, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. These might already address your question.

If you are asking about training for a custom instance segmentation task with YOLOv12, please provide more details about your setup and question. For example:
- Are you using a specific dataset? If so, any details (e.g., annotations, format, and number of images)?
- Have you started training with an existing [Ultralytics model](https://docs.ultralytics.com/models/) or are you looking to modify the architecture?

If this is a bug 🐛, we kindly ask that you provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help our team debug any issue more effectively.

## Upgrade

As a first step, ensure you're on the latest `ultralytics` version with all updated dependencies (including [PyTorch>=1.8](https://pytorch.org/get-started/locally/)) in a [Python>=3.8](https://www.python.org/) environment to access all the latest features:

```bash
pip install -U ultralytics
```

## Explore Environments

YOLO models, including tasks like instance segmentation, can be run in these verified environments:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Cloud Platforms** like [Google Cloud](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) or [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Dockers** for flexible configurations: See the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) or [Ultralytics Docker Hub](https://hub.docker.com/r/ultralytics/ultralytics) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Community & Support

Engage with the Ultralytics community in real-time on [Discord](https://discord.com/invite/ultralytics) 🎧, visit [Discourse](https://community.ultralytics.com/) for in-depth discussions, or explore our [Subreddit](https://reddit.com/r/Ultralytics) to exchange ideas.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>  
All [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests run regularly to ensure stability across all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu.

This is an automated response 🤖, but an Ultralytics engineer will join in to provide more assistance soon. We’re here to help! 😊

**Comment by Y-T-G** [comment]
You can use `model=yolo12-seg.yaml`. It will train from scratch since there are no pretrained weights for segmentation as of now.
